ingest/src/main/java/com/microsoft/azure/kusto/ingest/Utils.java

```diff
@@ -0,0 +1,110 @@
+package com.microsoft.azure.kusto.ingest;
+
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.storage.StorageException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.invoke.MethodHandles;
+import java.net.URISyntaxException;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+public class Utils {
+
+    private static final int COMPRESSED_FILE_MULTIPLIER = 11;
+    private final static Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    public static long estimateBlobRawSize(AzureStorageClient azureStorageClient, String blobPath) throws StorageException, URISyntaxException {
+        long blobSize = azureStorageClient.getBlobSize(blobPath);
+
+        return azureStorageClient.isCompressed(blobPath) ?
+                blobSize * COMPRESSED_FILE_MULTIPLIER : blobSize;
+    }
+
+    public static long estimateFileRawSize(AzureStorageClient azureStorageClient, String filePath) {
```

Comment:
can the azure client check if local files are compressed?

Reply:
Its just checks the path suffix:
return fileName.endsWith(".gz") || fileName.endsWith(".zip");